nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -267,15 +274,11 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
   // Find look ahead distance and point on path and publish
   double lookahead_dist = getLookAheadDistance(speed);
 
-  // Check for reverse driving
```

Comment:
I think now that the cusp distance is not being used in `transformGlobalPlan`, we should be able to revert this so that cusps are only worried about when `allow_reversing_` is true? Or is there a reason to keep it like this @sathak93 ?

Reply:
I don't understand the comment, sorry, can you elaborate?

Reply:
@sathak93 made this original change here to limit the lookahead distance to the closest cusp, regardless of whether reversing is enabled or not. I'm not sure if there is still a justification for this change: should we revert it to how it was?

Reply:
sorry for late reply. i think it is essential to make the robot to fully reach the pose point, when the path ahead  and the path behind the pose is overlapping.

Reply:
is this tested for overlapping case?

Reply:
> So, for a while, I was thinking that there were _two_ root causes of the problem: the starting pose selection for the transformation _and_ the lookahead point selection on the path wrapping around. I now recognize that only the first is actually a problem, so I don't need to worry about the cases where the plan loops around, as long as I don't choose that for the start of the transform.

for the overlapping path case lookahead point selection will surely cause the problem, it is still required to switch to integrated path distance

Reply:
I think it won't be a problem since the lookahead point selection uses the first point outside the lookhead distance. Even if the path loops back around, and even overlaps, those poses right by the robot shouldn't be selected, right?

Reply:
So I think this is in the same vein as the other comment that is still pending. If I understand this correctly:
- @Aposhian thinks we can remove this since we're no longer limited 
- @sathak93 wants to keep this so that the robot always gets to exactly the cusp point and then turns around to fully track the path

I think I agree with @Aposhian for now to have it cut the path short in that situation if not reversing. I think that is reasonably expected behavior for a non-feasible transition point to have a little shortness applied. But, I would also be open to adding a parameterization to this to enable it either way but I think that should be as part of another PR to not block this work from resolving 80% of the problem today. If @sathak93 you're open to submitting a PR to make a parameter to enable exact-full-path behavior, I'd be happy to review / merge it at a later date.

Reply:
@SteveMacenski I am open to follow up on this and contribute.